aws/resource_aws_s3_bucket_inventory.go

```diff
@@ -0,0 +1,445 @@
+			log.Printf("[WARN] %s S3 bucket inventory configuration not found, removing from state.", d.Id())
+			d.SetId("")
+			return nil
+		}
+		return err
+	}
+
+	d.Set("enabled", *output.InventoryConfiguration.IsEnabled)
+	d.Set("included_object_versions", *output.InventoryConfiguration.IncludedObjectVersions)
+	d.Set("optional_fields", flattenStringList(output.InventoryConfiguration.OptionalFields))
+	d.Set("filter", flattenS3InventoryFilter(output.InventoryConfiguration.Filter))
+	d.Set("schedule", flattenS3InventorySchedule(output.InventoryConfiguration.Schedule))
+	d.Set("destination", []map[string]interface{}{
+		{
+			"bucket": flattenS3InventoryS3BucketDestination(output.InventoryConfiguration.Destination.S3BucketDestination),
+		},
+	})
+
+	return nil
+}
+
+func expandS3InventoryFilter(m map[string]interface{}) *s3.InventoryFilter {
+	v, ok := m["prefix"]
+	if !ok {
+		return nil
+	}
+	return &s3.InventoryFilter{
+		Prefix: aws.String(v.(string)),
+	}
+}
+
+func flattenS3InventoryFilter(filter *s3.InventoryFilter) []map[string]interface{} {
+	if filter == nil {
+		return nil
+	}
+
+	result := make([]map[string]interface{}, 0, 1)
+
+	m := make(map[string]interface{}, 0)
+	if filter.Prefix != nil {
+		m["prefix"] = *filter.Prefix
+	}
+
+	result = append(result, m)
+
+	return result
+}
+
+func flattenS3InventorySchedule(schedule *s3.InventorySchedule) []map[string]interface{} {
+	result := make([]map[string]interface{}, 0, 1)
+
+	m := make(map[string]interface{}, 1)
+	m["frequency"] = *schedule.Frequency
+
+	result = append(result, m)
+
+	return result
+}
+
+func expandS3InventoryS3BucketDestination(m map[string]interface{}) *s3.InventoryS3BucketDestination {
+	destination := &s3.InventoryS3BucketDestination{
+		Format: aws.String(m["format"].(string)),
+		Bucket: aws.String(m["bucket_arn"].(string)),
+	}
+
+	if v, ok := m["account_id"]; ok && v.(string) != "" {
+		destination.AccountId = aws.String(v.(string))
+	}
+
+	if v, ok := m["prefix"]; ok && v.(string) != "" {
+		destination.Prefix = aws.String(v.(string))
+	}
+
+	if v, ok := m["encryption"].([]interface{}); ok && len(v) > 0 {
+		encryptionMap := v[0].(map[string]interface{})
+
+		encryption := &s3.InventoryEncryption{}
+
+		for k, v := range encryptionMap {
+			data := v.([]interface{})
+
+			if len(data) == 0 {
+				continue
+			}
+
+			switch k {
+			case "sse_kms":
+				m := data[0].(map[string]interface{})
+				encryption.SSEKMS = &s3.SSEKMS{
+					KeyId: aws.String(m["key_id"].(string)),
+				}
+			case "sse_s3":
+				encryption.SSES3 = &s3.SSES3{}
+			}
+		}
+
+		destination.Encryption = encryption
+	}
+
+	return destination
+}
+
+func flattenS3InventoryS3BucketDestination(destination *s3.InventoryS3BucketDestination) []map[string]interface{} {
+	result := make([]map[string]interface{}, 0, 1)
+
+	m := map[string]interface{}{
+		"format":     *destination.Format,
```

Comment:
Same with these -- to prevent potential panics, we should prefer to use the SDK provided function for dereferencing values (`aws.StringValue(destination.Format)`) instead of directly dereferencing via `*`.